src/db.rs

```diff
@@ -215,15 +215,27 @@ impl DbInner {
         let current_table = if self.wal_enabled() {
             let mut guard = self.state.write();
             let current_wal = guard.wal();
-            current_wal.put(key, value);
+            current_wal.put(
```

Comment:
> I'm envisioning a large 20+ file change PR every time we add a new attribute. 
> ...
> or whether we want it in some kind of cleaner struct that contains the row attrs and their values.

I'm not totally following, the idea of `RowAttributes` is to do exactly this (wrap it in a struct). I had originally written this PR passing the timestamp all the way down and then realized I'd need to do exactly the same thing when I added the TTL so I wrapped it in a struct `RowAttributes` that we can add to if we every want to add something else and not pipe it all the way through?

Reply:
> Right, but we'd still need to explicitly set the new property at every place where we create an instance of RowAttributes

This.

Reply:
Got it, that's not _too_ many places in production code (I specifically made all the tests call `gen_attr` instead of creating one manually so that we can add defaults for tests only). It seems like for now we're aligned that no defaults is OK, but we can revisit that later if it does make sense as we add attributes.

Reply:
imho it looks ok to pass Attributes everywhere inside the db impl, but when it's an API exposed to users, it'd be better to wrap the arguments inside something like a `WriteOptions` (https://docs.rs/rocksdb/latest/rocksdb/struct.WriteOptions.html), and better to avoid exposing the low-level encoding details on keys.

Reply:
Agreed, that was the intention! I don't think I exposed it anywhere in the public API, but I'll double check the PR to make sure.

Reply:
Gotcha. The `WriteOptions` / `RowAttributes` split makes total sense to me now.